src/vs/platform/webview/electron-main/webviewMainService.ts

```diff
@@ -3,17 +3,22 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { session, WebContents, webContents } from 'electron';
+import { session, WebContents, webContents, WebFrameMain } from 'electron';
+import { Emitter } from 'vs/base/common/event';
 import { Disposable } from 'vs/base/common/lifecycle';
 import { ITunnelService } from 'vs/platform/remote/common/tunnel';
-import { IWebviewManagerService, webviewPartitionId, WebviewWebContentsId, WebviewWindowId } from 'vs/platform/webview/common/webviewManagerService';
+import { FoundInFrameResult, IWebviewManagerService, webviewPartitionId, WebviewWebContentsId, WebviewWindowId } from 'vs/platform/webview/common/webviewManagerService';
 import { WebviewProtocolProvider } from 'vs/platform/webview/electron-main/webviewProtocolProvider';
 import { IWindowsMainService } from 'vs/platform/windows/electron-main/windows';
 
 export class WebviewMainService extends Disposable implements IWebviewManagerService {
 
 	declare readonly _serviceBrand: undefined;
 
+	private frameHandlers = new Map<string, WebFrameMain>();
```

Comment:
Does this need to be a map? I don't see cleanup logic for deleting entries when a webview is disposed of so I'm worried the `WebFrameMain` could be leaked . You may be able to avoid if this was a `set` (or, even better, is that a way you can avoid having state all together?)

Reply:
Changed around the logic to not require a map here.